conu/apidefs/image.py

```diff
@@ -170,6 +175,53 @@ def create_container(self, container_params):
         """
         raise NotImplementedError("create_container method is not implemented")
 
+    def run_in_pod(self, namespace="default"):
+        """
+        run image inside Kubernetes Pod
+        :param namespace: str, name of namespace where pod will be created
+        :return: Pod instance
+        """
+        config.load_kube_config()
+        api = client.CoreV1Api()
+
+        image_data = self.get_metadata()
+
+        # convert environment variables to Kubernetes objects
+        env_variables = []
+        for key, value in image_data.env_variables.items():
+            env_variables.append(client.V1EnvVar(name=key, value=value))
+
+        # convert exposed ports to Kubernetes objects
+        exposed_ports = []
+        if image_data.exposed_ports is not None:
+            for port in image_data.exposed_ports:
+                try:
+                    protocol = port.split("/", 1)[1]
+                    exposed_ports.append(client.V1ContainerPort(container_port=port.split("/", 1)[0],
+                                                                protocol=protocol))
+                except IndexError:  # protocol is not defined in image metadata
+                    exposed_ports.append(client.V1ContainerPort(container_port=port.split("/", 1)[0]))
+
+        # generate random container name
+        # https://stackoverflow.com/questions/2257441/random-string-generation-with-upper-case-letters-and-digits-in-python
+        container_name = ''.join(random.choice(string.ascii_lowercase) for _ in range(10))
+
+        container = client.V1Container(command=image_data.command,
+                                       env=env_variables,
+                                       image=image_data.name,
+                                       name=container_name,
+                                       ports=exposed_ports)
+
+        pod_metadata = client.V1ObjectMeta(name=container_name + "-pod")
+        pod_spec = client.V1PodSpec(containers=[container])
+        pod = client.V1Pod(spec=pod_spec, metadata=pod_metadata)
+
+        pod_instance = api.create_namespaced_pod(namespace=namespace, body=pod)
```

Comment:
There is an error when using container with exposed ports, e.g. `docker.io/nginx`

Log:
```
kubernetes.client.rest.ApiException: (400)
Reason: Bad Request
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'Date': 'Wed, 30 May 2018 12:19:27 GMT', 'Content-Length': '510'})
HTTP response body: {
"kind":"Status",
"apiVersion":"v1",
"metadata":{},
"status":"Failure",
"message":"Pod in version \"v1\" cannot be handled as a Pod: v1.Pod: Spec:
v1.PodSpec: Containers: []v1.Container: v1.Container: Ports:
[]v1.ContainerPort: v1.ContainerPort: ContainerPort:
readUint32: unexpected character: \ufffd, error found in #10 byte of ...|erPort\": \"80\",
\"prot|..., bigger context ...|name\": \"figpwbxmbt\", \"ports\":
[{\"containerPort\": \"80\", \"protocol\": \"tcp\"}]}]}}|...",
"reason":"BadRequest","code":400}

```